### PR TITLE
Added several error messages when trying to create an element that already exists

### PR DESCRIPTION
**Pull request by Sigurd-Borge**
Also refactored `create_study_local` to not need a "LocalConfiguration" object but just the path to the parent directory as a string.

### CODE REVIEW

File: src/antares/service/local_services/binding_constraint_local.py [hunk 1]
```diff
@@ -47,6 +48,10 @@ def create_binding_constraint(
         equal_term_matrix: Optional[pd.DataFrame] = None,
         greater_term_matrix: Optional[pd.DataFrame] = None,
     ) -> BindingConstraint:
+        if name in self.binding_constraints:
```
Comment (MartinBelthle) [inline src/antares/service/local_services/binding_constraint_local.py:51]: You should check with the id as it's what you store inside self.binding_constraints

File: src/antares/service/local_services/link_local.py [hunk 2]
```diff
@@ -74,15 +75,19 @@ def create_link(
...
                 properties_ini.read_file(ini_file)
         try:
             properties_ini.add_section(area_to.name)
-        except configparser.DuplicateSectionError as e:
-            raise CustomError(f"Link exists already, section already exists in properties.ini:\n\n{e.message}")
+        except DuplicateSectionError as e:
+            raise LinkCreationError(
+                area_from=area_from.name,
+                area_to=area_to.name,
+                message=f"Link exists already between '{area_from.name}' and '{area_to.name}', section already exists in properties.ini:\n\n{e.message}",
```
Comment (MartinBelthle) [inline src/antares/service/local_services/link_local.py:89]: The first part of the msg is enough I believe: f"Link exists already between '{area_from.name}' and '{area_to.name}'"

File: src/antares/service/local_services/area_local.py [hunk 3]
```diff
@@ -84,7 +84,14 @@ def create_thermal_cluster(
...
 
         list_ini = IniFile(self.config.study_path, IniFileTypes.THERMAL_LIST_INI, area_name=area_id)
-        list_ini.add_section(local_thermal_properties.list_ini_fields)
+        try:
+            list_ini.add_section(local_thermal_properties.list_ini_fields)
+        except DuplicateSectionError as error:
+            raise ThermalCreationError(
+                thermal_name,
+                area_id,
+                f"A thermal cluster called '{thermal_name}' already exists in area '{area_id}'. " + error.message,
```
Comment (MartinBelthle) [inline src/antares/service/local_services/area_local.py:93]: Same here don't need the errror.message I think

File: src/antares/model/study.py [hunk 4]
```diff
@@ -77,7 +78,7 @@ def create_study_api(
 def create_study_local(
     study_name: str,
     version: str,
-    local_config: LocalConfiguration,
+    parent_directory: str,
```
Comment (MartinBelthle) [inline src/antares/model/study.py:81]: I think it's better indeed

File: src/antares/model/study.py [hunk 5]
```diff
@@ -216,6 +218,8 @@ def get_binding_constraints(self) -> MappingProxyType[str, BindingConstraint]:
     def create_area(
         self, area_name: str, *, properties: Optional[AreaProperties] = None, ui: Optional[AreaUi] = None
     ) -> Area:
+        if area_name in self._areas and isinstance(self._study_service, StudyLocalService):
```
Comment (MartinBelthle) [inline src/antares/model/study.py:221]: Should compare id and not name. And it should be done inside the local service method create_area, not here as you introduced the notion of local inside a model which isn't supposed to have one (plus it's a bit ugly)

File: tests/antares/services/local_services/test_area.py [hunk 6]
```diff
@@ -424,7 +444,7 @@ def test_st_storage_list_ini_has_correct_default_values(
         assert actual_st_storage_list_ini.parsed_ini.sections() == expected_st_storage_list_ini.sections()
         assert actual_st_storage_list_ini.parsed_ini == expected_st_storage_list_ini
 
-    def test_st_storage_and_ini_have_custom_properties(self, local_study_with_st_storage, actual_st_storage_list_ini):
+    def test_st_storage_and_ini_have_custom_properties(self, local_study_with_renewable):
```
Comment (MartinBelthle) [inline tests/antares/services/local_services/test_area.py:447]: Really weird to use _with_renewable in a st_storage test, why did you change this ?

File: tests/antares/services/local_services/test_area.py [hunk 7]
```diff
@@ -52,6 +55,18 @@ def test_can_be_created(self, local_study_w_areas):
         created_thermal = local_study_w_areas.get_areas()["fr"].create_thermal_cluster(thermal_name)
         assert isinstance(created_thermal, ThermalCluster)
 
+    def test_duplicate_name_errors(self, local_study_w_thermal):
```
Comment (MartinBelthle) [inline tests/antares/services/local_services/test_area.py:58]: Duplicated test inside test_study. This one should be removed i think

**Review by MartinBelthle — CHANGES_REQUESTED** [review]
Few changes requested

File: tests/antares/services/local_services/test_area.py (hunk 6 above)
Comment (Sigurd-Borge) [reply tests/antares/services/local_services/test_area.py:447]: Because in this test I test that creating a short term storage with custom properties work, this way it's clear that there is no st storage in the area yet and avoids potential duplicate st storage error.

Comment (MartinBelthle) [reply tests/antares/services/local_services/test_area.py:447]: Okay so it would be better to use _w_areas just to make it clearer

**Review by vargastat — APPROVED** [review]

File: tests/antares/services/local_services/test_area.py (hunk 7 above)
Comment (Sigurd-Borge) [reply tests/antares/services/local_services/test_area.py:58]: I'm a bit confused, I don't see a dupe and this seems to be the right place to test this as thermals are part of areas. Could you point me at the duplicated test please?

File: tests/antares/services/local_services/test_area.py (hunk 6 above)
Comment (Sigurd-Borge) [reply tests/antares/services/local_services/test_area.py]: Corrected

File: src/antares/service/local_services/binding_constraint_local.py [hunk 8]
```diff
@@ -47,6 +49,10 @@ def create_binding_constraint(
         equal_term_matrix: Optional[pd.DataFrame] = None,
         greater_term_matrix: Optional[pd.DataFrame] = None,
     ) -> BindingConstraint:
+        if transform_name_to_id(name) in self.binding_constraints:
```
Comment (MartinBelthle) [inline src/antares/service/local_services/binding_constraint_local.py:52]: Mmmmh you assume the id equals transform_name_to_id(name) but it's an implementation detail of the class BindingConstraint. You should just put this code after creating the constraint as you can use the created id without assuming anything

File: tests/antares/services/local_services/test_area.py (hunk 7 above)
Comment (MartinBelthle) [reply tests/antares/services/local_services/test_area.py:58]: You're right I misread my bad

**Review by MartinBelthle — CHANGES_REQUESTED** [review]
Just one minor change and we're good to go :)

**Review by MartinBelthle — APPROVED** [review]
